### ISSUE TITLE
Invalid Refaster suggestion because parentheses are not added

### ISSUE THREAD

**Issue by PhilippWendler** [google/error-prone#4866]
I want to use the following Refaster rule:

```java
class StringFormat {

  @BeforeTemplate
  String format(String format, @Repeated Object more) {
    return String.format(format, more);
  }

  @AfterTemplate
  String formatted(String format, @Repeated Object more) {
    return format.formatted(more);
  }
}
```

However, when applied to code where `format` is an expression, not just a literal, the suggestion replacement becomes invalid because Refaster fails to add parentheses around `format` when they are necessary.
Example:
```java
public class Test {

  public static void main(String[] args) {
    System.out.println(String.format("%s " + "another string", "test"));
  }
}
```
is rewritten to
```java
public class Test {

  public static void main(String[] args) {
    System.out.println("%s " + "another string".formatted("test"));
  }
}
```
But it needs to be `("%s " + "another string").formatted("test")`. I can manually add parentheses to `@AfterTemplate`, but then I get them all the time, even when not necessary.

I think that Refaster already adds parentheses in other cases if required, but the case of a receiver of a method call seems to be unhandled.

**Comment by cushon** [comment]
I think this might be a dupe of #4819, which was just fixed.

@graememorgan is also working on some additional improvements that should make Inliner more robust.

**Comment by graememorgan** [comment]
Is it a dupe? #4819 is `@InlineMe`, while this is refaster.

**Comment by cushon** [comment]
Sorry, I was confused earlier, as Graeme said in https://github.com/google/error-prone/issues/4866#issuecomment-2701584213 the fix I was thinking of was for InlineMe, not for refaster

**Comment by PhilippWendler** [comment]
Is there a chance that this could be fixed for Refaster as well? The rule above would be a nice thing to have for modern Java projects.
